draft-ietf-quic-transport.md

```diff
@@ -1451,29 +1454,276 @@ failure.  If integrity protection is performed by QUIC, QUIC MUST abort the
+packets.
+
+
+### Probing a New Path
+
+An endpoint MAY probe for peer reachability from a new local address using path
+validation {{migrate-validate}} prior to migrating the connection to the new
+local address.  Failure of path validation simply means that the new path is not
+usable for this connection.  Failure to validate a path does not cause the
+connection to end unless there are no valid alternative paths available.
+
+An endpoint uses a new connection ID for probes sent from a new local address,
+see {{migration-linkability}} for further discussion.
+
+Receiving a PATH_CHALLENGE frame from a peer indicates that the peer is probing
+for reachability on a path. An endpoint sends a PATH_RESPONSE in response as per
+{{migrate-validate}}.
+
+PATH_CHALLENGE, PATH_RESPONSE, and PADDING frames are "probing" frames. Until an
+endpoint receives packets containing other frames from a given peer address, it
+MUST NOT send other types of frames to that address.
+
+
+### Initiating Connection Migration {#initiating-migration}
+
+A endpoint can migrate a connection to a new local address by sending packets
+containing frames other than probing frames from that address.
 
-An endpoint that receives packets that contain a source IP address and port that
-has not yet been used can start sending new packets with those as a destination
-IP address and port.  Packets exchanged between endpoints can then follow the
-new path.
+Each endpoint validates its peer's address during connection establishment.
+Therefore, a migrating endpoint can send to its peer knowing that the peer is
+willing to receive at the peer's current address. Thus an endpoint can migrate
+to a new local address without first validating the peer's address.
 
-Due to variations in path latency or packet reordering, packets from different
-source addresses might be reordered.  The packet with the highest packet number
-MUST be used to determine which path to use.  Endpoints also need to be prepared
-to receive packets from an older source address.
+When migrating, the new path might not support the endpoint's current sending
+rate. Therefore, the endpoint resets its congestion controller, as described in
+{{migration-cc}}.
 
-An endpoint MUST validate that its peer can receive packets at the new address
-before sending any significant quantity of data to that address, or it risks
-being used for denial of service.  See {{migrate-validate}} for details.
+Receiving acknowledgments for data sent on the new path serves as proof of the
+peer's reachability from the new address.  Note that since acknowledgments may
+be received on any path, return reachability on the new path is not
+established. To establish return reachability on the new path, an endpoint MAY
+concurrently initiate path validation {{migrate-validate}} on the new path.
+
+
+### Responding to Connection Migration {#migration-response}
+
+Receiving a packet from a new peer address containing a non-probing frame
```

Comment:
If a packet contains a PATH_CHALLENGE and a STREAM frame, then it contains a non-probing frame, but I don't think the intent is to immediately migrate.

How about "Receiving a packet from a new peer address that does not contain a PATH_CHALLENGE frame indicates the peer has migrated to that address."

Reply:
I think that's a substantive change to the design.  If you don't intend to migrate, you're not using that path for anything except probing.  If you're including a STREAM frame, then you are migrating, though you might also want to explicitly probe to be sure the migration is working.

Reply:
Above the text says it's ok to bundle other frames with PATH_CHALLENGE, so I assumed STREAM frames were ok.  Is that not true?  Whatever we decide, this should be clear.

Reply:
It's ok to bundle other frames, as the server might, when it's verifying client ownership of the new address. But if a client sends anything but probing frames to the server, the server considers this as a migration, not a probe.